src/webui/api/appcontroller.cpp

```diff
@@ -1112,6 +1114,40 @@ void AppController::sendTestEmailAction()
     app()->sendTestEmail();
 }
 
+
+void AppController::getDirectoryContentAction()
+{
+    requireParams({u"dirPath"_s});
+
+    const QString dirPath = params().value(u"dirPath"_s);
+    if (dirPath.isEmpty() || dirPath.startsWith(u':'))
+        throw APIError(APIErrorType::BadParams, QObject::tr("Invalid directory path"));
+
+    const QDir dir {dirPath};
+    if (!dir.isAbsolute())
+        throw APIError(APIErrorType::BadParams, QObject::tr("Invalid directory path"));
+    if (!dir.exists())
+        throw APIError(APIErrorType::NotFound, QObject::tr("Directory does not exist"));
+
+    const QString visibility = params().value(u"mode"_s, u"all"_s);
+
+    const auto parseDirectoryContentMode = [](const QString &visibility)
```

Comment:
We require explicit return type:
```suggestion
    const auto parseDirectoryContentMode = [](const QString &visibility) -> QDir::Filters
```

Reply:
> We require explicit return type

Really?
Although in this case it makes sense.